### PR TITLE
EEI: blockhash to signal success or failure

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by jwasinger** [comment]
As an alternative, we could have the behavior`BLOCKHASH(num>256) = 0x0...0` in order to be more EVM-compliant.

However, this proposed solution is more transparent IMO.

**Comment by axic** [comment]
We are trying to avoid the implicit reporting. Also hope that before eWASM is final, the blockhash EIP will be merged and we could remove this interface.

File: eth_interface.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ Gets the hash of one of the 256 most recent complete blocks.
 
 **Returns**
 
-*nothing*
+`result` **i32** Returns 1 or 0 depending on the operation succeeded or not
```
Comment (chfast) [inline eth_interface.md]: Does 0 mean success?

Comment (axic) [reply eth_interface.md]: Agree this should be clarified (clarified the ones for calls in #77).

We can either go POSIX style and have 0 as success (matches #77) or with a boolean where 1 is success (matches evm).

Comment (chfast) [reply eth_interface.md]: I think we regret having only single failure flag in EVM 1.0. I would go with POSIX as in #77. I'm sure we will have proper discussion about this when a draft will be published.

Comment (jakelang) [reply eth_interface.md]: I wonder why EVM deviated from POSIX in the first place.

**Comment by axic** [comment]
The other question is shouldn't al the other non-returning but data writing methods have a return value? e.g. `codeCopy`, `externalCodeCopy`, etc.

**Comment by chfast** [comment]
Probably they should. VM should also has an information whenever an account exists.

**Comment by axic** [comment]
Updated to march #77 and added a comment regarding the output memory.

**Pull request merged**